### PR TITLE
 Repaired header file include guards in boards directory #2623

### PR DESCRIPTION
**Pull request by shrenikjain38**
The include guards in boards directory have been repaired. Also the defines beginning with a single underscore are repaired. Require further clarification with reference to defines beginning with single underscore followed by an uppercase letter.

### CODE REVIEW

**Comment by authmillenon** [comment]
Fixes #2623

File: boards/msb-430/include/board.h [hunk 1]
```diff
@@ -33,8 +33,8 @@ extern "C" {
 #endif
 
 // for correct inclusion of <msp430.h>
-#ifndef __MSP430F1612__
-#define __MSP430F1612__
+#ifndef MSP430F1612__
+#define MSP430F1612__
```
Comment (gebart) [inline boards/msb-430/include/board.h]: This probably breaks msp430, see the comment on L35

File: boards/msba2-common/tools/src/serial.c [hunk 2]
```diff
@@ -355,7 +355,7 @@ int set_baud(const char *baud_name)
         return -3;
     }
 
-#ifdef __APPLE__
+#ifdef APPLE__
```
Comment (gebart) [inline boards/msba2-common/tools/src/serial.c]: This will break OSX

**Comment by gebart** [comment]
You changed more than just the include guards and broke a few tools and platforms in the process.

**Comment by haukepetersen** [comment]
needs rebase

**Comment by gebart** [comment]
closing because of inactivity

**Pull request closed without merging**